src/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.cs

```diff
@@ -162,6 +162,33 @@ public void MaxRequestContentBufferSize_Set_ThrowsPlatformNotSupportedException(
             }
         }
 
+        [Theory]
+        [MemberData(nameof(GetAsync_IPv6Uri_Success_MemberData))]
+        public async Task GetAsync_IPv6Uri_Success(IPAddress address)
+        {
+            using (var client = new HttpClient())
+            {
+                var options = new LoopbackServer.Options { Address = address };
+                await LoopbackServer.CreateServerAsync(async (server, url) =>
+                {
+                    await TestHelper.WhenAllCompletedOrAnyFailed(
```

Comment:
> Is this the new suggested pattern for writing tests using the loopback server, i.e. this  TestHelper.WhenAllCompletedOrAny  method?

I'm experimenting with it to see how well it works; so far so good.  The idea here is that it lets us ensure we've cleaned up after all of the resources but that we don't end up deadlocking if, for example, the client fails to connect and the server just sits there waiting for a connection indefinitely.